### PR TITLE
Actor Memory

### PR DESCRIPTION
**Pull request by racerpeter**
Add the ability to remember where we left off by persisting the next binlog position by way of a "saver" actor as the final step in the process.

Save the last known position periodically in case of a crash/hardware failure, and also cleanly persist the last emitted position on shutdown.

- [x] Finish wiring up integration tests
- [x] Green/Green cleanup/refactor
- [ ] Test in staging
- [ ] Bonus points for gtid support

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 changestream {
   mysql {
     host = "localhost"
```
Comment (bmidgley) [inline src/main/resources/application.conf]: Is it also required or advisable to change the host to `host = "mysql.mavenlink.link"` ?

File: src/main/resources/application.conf [hunk 2]
```diff
@@ -142,6 +142,14 @@ changestream {
   control.host = ${?CONTROL_HOST}
   control.port = 9521
   control.port = ${?CONTROL_PORT}
+
+  // File path for position saver
+  position-saver.file-path = "/tmp/changestream.pos"
```
Comment (mavenky) [inline src/main/resources/application.conf]: Seems like we are saving the location in a file. What happens if we lose changestream server?

Comment (racerpeter) [reply src/main/resources/application.conf]: Great question @mavenky -- this particular position saver is set up to save to a file, but you could write a saver that saved to anywhere you wanted.

In our use case, the instance changestream is running on is EBS-backed, so no problems with losing data.

**Comment by racerpeter** [comment]
https://github.com/mavenlink/changestream/issues/4

**Comment by racerpeter** [comment]
https://github.com/mavenlink/changestream/issues/38

**Pull request merged**